Simplify and optimize Hilbert tile ID <-> XYZ conversion

ciscorn opened the pull request:
I found that we can simplify and optimize the conversion between XYZ tile coordinates and the Hilbert tile IDs.

This optimization is primarily based on the fact that the tile IDs for (zi, 0, 0) are given by the sum of a geometric series with a common ratio of 4.

Related PR: https://github.com/protomaps/go-pmtiles/pull/208

bdon commented:
Same with https://github.com/protomaps/PMTiles/pull/470 - please provide some evidence or benchmark this is faster across different JavaScript engines, otherwise we will just have other PRs optimizing it in different ways (iterative, lookup table, etc)

ciscorn commented:
Sorry for unintentionally creating a duplicate PR. I’ll take some time later to run benchmarks.

ciscorn commented:
For now, I benchmarked this in the Go implementation (See: https://github.com/protomaps/go-pmtiles/pull/208).

Of course, this improvement may not necessarily apply to JavaScript or other languages.

ciscorn commented:
I have now benchmarked the JavaScript implementation across multiple runtimes and browsers.

Benchmarking script: https://gist.github.com/ciscorn/e84d41fe5c7ac7c1857d8f64fc1df069

Results on my M1 Apple Silicon:

Chrome 133

```
zxyToTileId: Original=2665.60(ms), Modified=42.50(ms) (62.72x faster)
tileIdToZxy: Original=548.10(ms), Modified=381.60(ms) (1.44x faster)
```

Firefox 125

```
zxyToTileId: Original=2677.00(ms), Modified=47.00(ms) (56.96x faster)
tileIdToZxy: Original=942.00(ms), Modified=555.00(ms) (1.70x faster)
```

NodeJS 22.12

```
zxyToTileId: Original=2760.29(ms), Modified=40.69(ms) (67.83x faster)
tileIdToZxy: Original=625.26(ms), Modified=384.92(ms) (1.62x faster)
```

Bun 1.2.2

```
zxyToTileId: Original=1483.75(ms), Modified=80.07(ms) (18.53x faster)
tileIdToZxy: Original=1003.57(ms), Modified=702.73(ms) (1.43x faster)
```

Deno 2.1.10

```
zxyToTileId: Original=2730.63(ms), Modified=40.57(ms) (67.30x faster)
tileIdToZxy: Original=646.00(ms), Modified=380.45(ms) (1.70x faster)
```

bdon commented:
Released in https://www.npmjs.com/package/pmtiles 4.3.0, thanks!
